com.unity.sg2/Editor/GraphDeltaRegistry/Registry/Registry.cs

```diff
@@ -108,6 +108,31 @@ internal ShaderFoundry.ShaderType GetShaderType(FieldHandler field, ShaderFoundr
         public IEnumerable<RegistryKey> BrowseRegistryKeys() => builders.Keys;
 
         public bool CastExists(RegistryKey from, RegistryKey to) => builders.Values.OfType<ICastDefinitionBuilder>().Any(e => e.GetTypeConversionMapping().Equals((from,to)));
+        public bool CastExists(PortHandler from, PortHandler to)
+        {
+            var possible = builders.Values.OfType<ICastDefinitionBuilder>().Where(e => e.GetTypeConversionMapping().Equals((from.GetRegistryKey(), to.GetRegistryKey())));
+            foreach(var p in possible)
+            {
+                if(p.CanConvert(from.GetTypeField(), to.GetTypeField()))
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        internal ICastDefinitionBuilder GetCast(PortHandler from, PortHandler to)
+        {
+            var possible = builders.Values.OfType<ICastDefinitionBuilder>().Where(e => e.GetTypeConversionMapping().Equals((from.GetRegistryKey(), to.GetRegistryKey())));
+            foreach (var p in possible)
+            {
+                if (p.CanConvert(from.GetTypeField(), to.GetTypeField()))
+                {
+                    return p;
+                }
+            }
+            return null;
+        }
```

Comment:
Should these be fields instead of ports? I guess automatic casting is only applicable for ports; but is there a situation where you can just have fields and want to check/get the cast for them?